src/cargo/ops/cargo_new.rs

```diff
@@ -925,3 +941,79 @@ fn update_manifest_with_inherited_workspace_package_keys(
         try_remove_and_inherit_package_key(key, manifest);
     }
 }
+
+/// Adds the new package member to the [workspace.members] array.
+/// - It first checks if the name matches any element in [workspace.exclude],
+///  and it ignores the name if there is a match.
+/// - Then it check if the name matches any element already in [workspace.members],
+/// and it ignores the name if there is a match.
+/// - If [workspace.members] doesn't exist in the manifest, the new package
+/// is not added to the members array. This prevents manifest files from breaking
+/// if they're only used for package templating.
+fn update_manifest_with_new_member(
+    manifest_path: &Path,
+    workspace_document: &mut toml_edit::Document,
+    relative_path: &str,
+) -> CargoResult<()> {
+    // Don't add the new package to the workspace's members
+    // if there is an exclusion match for it.
+    if let Some(members) = workspace_document
+        .get("workspace")
+        .and_then(|workspace| workspace.get("exclude"))
+        .and_then(|members| members.as_array())
+    {
+        let members = members.iter().map(|s| s.as_str().unwrap());
+        if matches_in_package_list(members, relative_path)? {
+            return Ok(());
+        }
+    }
+
+    if let Some(members) = workspace_document
+        .get("workspace")
+        .and_then(|workspace| workspace.get("members"))
+        .and_then(|members| members.as_array())
+    {
+        let members = members.iter().map(|s| s.as_str().unwrap());
+        if !matches_in_package_list(members, relative_path)? {
+            let Some(toml_members) = workspace_document["workspace"]["members"].as_array_mut()
+            else {
+                anyhow::bail!("[workspace.members] is not an array");
+            };
+            toml_members.push(relative_path);
+            paths::write(&manifest_path, workspace_document.to_string())?;
+        }
+    }
+
+    Ok(())
+}
+
+fn matches_in_package_list<'a, P: Iterator<Item = &'a str>>(
+    members: P,
+    package_path: &str,
+) -> CargoResult<bool> {
+    for name in members {
+        let pattern = build_glob(name)?;
+        if pattern.matches(package_path) {
```

Comment:
`package_path` -> `relative_path` which is the path relative to the CWD which is different than the path relative to the workspace root.
1. Please add a test for this case
2. Please make the test path.  One option is to use `pathdiff` to find the path of the new package relative to the manifest root that was discovered.

Reply:
See https://github.com/rust-lang/cargo/pull/12779/files#diff-90bbae39fa79149d8bb0faceba6b32e93efbe939964f77c3ad5077a3d0a86942R810